src/doc_table.c

```diff
@@ -9,24 +9,76 @@
 #include "rmalloc.h"
 #include "spec.h"
 
+#define MAX_SIZE 1000000
+
 /* Creates a new DocTable with a given capacity */
 DocTable NewDocTable(size_t cap) {
   return (DocTable){.size = 1,
                     .cap = cap,
                     .maxDocId = 0,
                     .memsize = 0,
                     .sortablesSize = 0,
-                    .docs = rm_calloc(cap, sizeof(RSDocumentMetadata)),
+                    .maxSize = MAX_SIZE,
+                    .buckets = rm_calloc(cap, sizeof(DMDChain)),
                     .dim = NewDocIdMap()};
 }
 
+static inline uint32_t DocTable_GetBucket(DocTable *t, t_docId docId) {
+  return docId < t->maxSize ? docId : docId % t->maxSize;
```

Comment:
I'm curious if the check and return is quicker or slower than simply doing mod always. Maybe this has to do with the growing algorithm?

Reply:
@dvirsky said that mod is an expensive operation, that's why we decided to do it this way.